File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    release = []
    for p, s in zip(progresses, speeds):
        div, mod = divmod(100 - p, s)
        release.append(div if mod == 0 else div + 1)
        
    stack, answer = [], []
    for r in release:
        if stack and stack[0] < r:
            answer.append(len(stack))
            stack.clear()
        stack.append(r)
    answer.append(len(stack))
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int n = progresses.length;
        
        int[] release = new int[n];
        int p, s, div, mod;
        for (int i = 0; i < n; i++) {
            p = progresses[i]; s = speeds[i];
            div = (int) (100 - p) / s; mod = (100 - p) % s;
            release[i] = (mod == 0) ? div : div + 1; 
        }
        
        Deque<Integer> q = new ArrayDeque<>();
        List<Integer> list = new ArrayList<>();
        for (int r : release) {
            if (!q.isEmpty() && q.peek() < r) {
                list.add(q.size());
                q.clear();
            }
            q.offer(r);
        }
        list.add(q.size());
        
        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}